src/topology/bounded_continuous_function.lean

```diff
@@ -429,5 +429,109 @@ def of_normed_group_discrete {α : Type u} {β : Type v}
   (f : α  → β) (C : ℝ) (H : ∀x, norm (f x) ≤ C) : α →ᵇ β :=
 of_normed_group f C H continuous_of_discrete_topology
 
+/- Formula for norm. -/
+lemma norm_eq ( f : α →ᵇ β) :
+  ∥f∥ = Inf {C : ℝ | 0 ≤ C ∧ ∀ (x : α), ∥f x∥ ≤ C} :=
+begin
+  rw ← sub_zero f,
+  rw ← dist_eq_norm,
+  rw bounded_continuous_function.dist_eq,
+  simp,
+end
+
+
 end normed_group
+
+
+section normed_space
+/- In this section, if β is a normed space, then we show that the space of bounded
+continuous functions from α to β inherits a normed space structure, by using
+pointwise operations and checking that they are compatible with the uniform distance. -/
+
+variables [nondiscrete_normed_field 𝕜]
+variables [topological_space α] [normed_group β] [normed_space 𝕜 β]
+variables {f g : α →ᵇ β} {x : α} {C : ℝ}
+
+
+instance : has_scalar 𝕜 (α →ᵇ β) :=
+  ⟨λc, λf, ⟨λx, c • f.1 x,
+    continuous.smul continuous_const f.2.left,
+    begin
+      cases f.2.right with C hbound,
+      use ∥c∥ * C,
+      intros,
+      have hnneg : 0 ≤ ∥c∥,
+      { exact norm_nonneg c },
+      specialize hbound x y,
+      rw dist_eq_norm at hbound,
+      rw dist_eq_norm,
+      calc ∥c • f x - c • f y∥ = ∥c • (f x - f y)∥ : by rw smul_sub c (f x) (f y)
+      ... = ∥c∥ * ∥f x - f y∥ : norm_smul c (f x - f y)
+      ... ≤ ∥c∥ * C : mul_le_mul_of_nonneg_left hbound hnneg,
+    end⟩⟩
+
+
+instance : module 𝕜 (α →ᵇ β) :=
+  module.of_core $ begin
+    refine { smul := (•), ..},
+    intros c f g, ext,
+    exact smul_add c (f x) (g x),
+    intros c₁ c₂ f, ext,
+    exact add_smul c₁ c₂ (f x),
+    intros c₁ c₂ f,
+    ext, exact mul_smul c₁ c₂ (f x),
+    intros f,
+    ext, exact one_smul 𝕜 (f x),
+  end
+
+
+instance : vector_space 𝕜 (α →ᵇ β) :=
+  { .. bounded_continuous_function.module }
+
+
+lemma bounded_continuous_sub_smul (c : 𝕜) (f : α →ᵇ β) :  ∥c • f∥ ≤ ∥c∥ * ∥f∥ :=
+begin
+  have hnneg : 0 ≤ ∥ c ∥,
+    exact norm_nonneg c,
+  rw norm_eq (c • f),
+  apply real.Inf_le,
+  { use 0, intros y hy,
+    rw set.mem_set_of_eq at hy,
+    exact hy.left,},
+  { rw set.mem_set_of_eq,
+    split,
+    exact mul_nonneg' hnneg (norm_nonneg f),
+    intros,
+    calc ∥(c • f) x∥ = ∥c∥ * ∥f x∥ : norm_smul c (f x)
+    ... ≤ ∥c∥ * ∥f∥ : mul_le_mul_of_nonneg_left (norm_coe_le_norm x) hnneg,}
+end
+
+
+lemma bounded_continuous_smul (c : 𝕜) (f : α →ᵇ β) :  ∥c • f∥ = ∥c∥ * ∥f∥ :=
+begin
+  by_cases ( c = 0),
+  rw h, simp,
```

Comment:
The `rw h` is referring to `h` which is an auto-generated name. We try to avoid that. So you could write `cases ... with h h` on the line above.

Reply:
fixed as `by_cases h : c = 0`